### ISSUE TITLE
`typing_extensions` incompatible with -OO

### ISSUE THREAD

**Issue by rkr-at-dbx** [python/typing#524]
If docstrings are stripped (as by the -OO flag), then under Py3, `typing_extensions` will crash at import time when it tries to alter `Protocol.__doc__`:

>```
>    Protocol.__doc__ = Protocol.__doc__.format(bases="Protocol, Generic[T]" if
>                                               OLD_GENERICS else "Protocol[T]")
>```

**Comment by ilevkivskyi** [comment]
Hm, indeed this case was missed. Fortunately this one is easy to fix by just checking if ``__doc__ is not None``, would you like to make a PR?

**Comment by Nickatak** [comment]
Hello! I'm a newcome to github (as a whole).  Do you guys mind if I make an attempt at fixing this issue (I'm still reading through the workflow/figuring out how to test a change I make with this particular repo [I'm no stranger to Python, but I've never contributed to it before])?  If not: it's okay, I understand how frustrating newcomers can be to anything, let alone something so complex. C:

**Comment by ethanhs** [comment]
Hello and welcome @Nickatak. I don't think anyone will stop you from working on this :) I am happy to help you if needed.

>Do you happen to have any windows instructions for testing?

There aren't to my knowledge but here is a sample (after downloading the repository).

```shell
\typing>pip install -r test-requirements.txt
\typing>pip install tox
\typing>tox
```

If you don't have all of the python interpreters requested in the tox config (2.7, 3.3, 3.4, 3.5, 3.6, 3.6) it will complain about this, but in this case you probably don't need to worry about that.

**Comment by Nickatak** [comment]
Interesting, thank you very much! C: I will attempt over the following few days and update you on my progress. :D

**Comment by Nickatak** [comment]
All tox tests run fine with the changes in the previous submitted PR; however, I still feel unsure about submitting a PR given that I am not able to reproduce the original issue (I don't really understand how the issue came to be, nor the -OO flag, sorry).  May I ask you how to run the -OO flag on a generic install (Windows 10, x64, Python==3.6.4) on a simple test-file so I may reproduce the issue, and then test to see if my changes fix the original issue? OR, can you provide me with a link to the -OO flag's documentation so I may further research (and report my findings)?  I do NOT want to submit a PR without validating that the original issue is resolved.

**Comment by ilevkivskyi** [comment]
This is now fixed by #527
